optimum/bettertransformer/models/distilbert/modeling_distilbert_fast.py

```diff
@@ -0,0 +1,122 @@
+        """
+        super().__init__()
+        # In_proj layer
+        self.in_proj_weight = nn.Parameter(
+            torch.cat(
+                [
+                    bert_layer.attention.q_lin.weight,
+                    bert_layer.attention.k_lin.weight,
+                    bert_layer.attention.v_lin.weight,
+                ]
+            )
+        )
+        self.in_proj_bias = nn.Parameter(
+            torch.cat(
+                [
+                    bert_layer.attention.q_lin.bias,
+                    bert_layer.attention.k_lin.bias,
+                    bert_layer.attention.v_lin.bias,
+                ]
+            )
+        )
+
+        # Out proj layer
+        self.out_proj_weight = bert_layer.attention.out_lin.weight
+        self.out_proj_bias = bert_layer.attention.out_lin.bias
+
+        # Linear layer 1
+        self.linear1_weight = bert_layer.ffn.lin1.weight
+        self.linear1_bias = bert_layer.ffn.lin1.bias
+
+        # Linear layer 2
+        self.linear2_weight = bert_layer.ffn.lin2.weight
+        self.linear2_bias = bert_layer.ffn.lin2.bias
+
+        # Layer norm 1
+        self.norm1_eps = bert_layer.sa_layer_norm.eps
+        self.norm1_weight = bert_layer.sa_layer_norm.weight
+        self.norm1_bias = bert_layer.sa_layer_norm.bias
+
+        # Layer norm 2
+        self.norm2_weight = bert_layer.output_layer_norm.weight
+        self.norm2_bias = bert_layer.output_layer_norm.bias
+
+        # Model hyper parameters
+        self.num_heads = bert_layer.attention.n_heads
+        self.embed_dim = bert_layer.attention.dim
+
+        del bert_layer
+
+        # Last step: set the last layer to `False` -> this will be set to `True` when converting the model
+        self.is_last_layer = False
+
+    def forward(self, x, attn_mask, head_mask=None, output_attentions=None, *_):
+        r"""
+        This is just a wrapper around the forward function proposed in:
+        https://github.com/huggingface/transformers/pull/19553
+        """
+        if x.is_nested:
+            attn_mask = None
+
+        if attn_mask is not None:
+            # attention mask comes in with values 0 and -inf. we convert to torch.nn.TransformerEncoder style bool mask
+            # 0->false->keep this token -inf->true->mask this token
+            attn_mask = attn_mask.bool()
+            attn_mask = torch.reshape(attn_mask, (attn_mask.shape[0], attn_mask.shape[-1]))
+            seqlen = attn_mask.shape[1]
+            lengths = torch.sum(~attn_mask, 1)
+            if not all([l == seqlen for l in lengths]):
+                x = torch._nested_tensor_from_mask(x, attn_mask)
+            attn_mask = None
+
+        x = torch._transformer_encoder_layer_fwd(
+            x,
+            self.embed_dim,
+            self.num_heads,
+            self.in_proj_weight,
+            self.in_proj_bias,
+            self.out_proj_weight,
+            self.out_proj_bias,
+            True,  # TODO use_gelu. make it not hardcoded
```

Comment:
Same comment about the exceptions.

Reply:
Fixed in a74192df94b8704f351cca6f87d192cb2433fdcd